src/utils/configuration.cpp

```diff
@@ -122,8 +123,15 @@ void ConfigurationMap::addKeys(keys_t& names) const
 bool ConfigurationEnviron::tryGetPropertyAsString(const std::string& name, std::string* val) const
 {
     const char *env = getenv(name.c_str());
-    if(!env || !*env)
+    if(!env || !*env) {
+		const ENV_PARAM *pParam;
+		pParam = envFindConfigParam( name.c_str() );
+		if ( pParam && pParam->pdflt ) {
+			*val = pParam->pdflt;
+			return true;
+		}
         return false;
+	}
```

Comment:
The logic here can be slightly simplified as:

```c++
bool ConfigurationEnviron::tryGetPropertyAsString(const std::string& name, std::string* val) const
{
    const char *env = getenv(name.c_str());
    if(!env) {
        const ENV_PARAM *pParam = envFindConfigParam( name.c_str() );
        if(pParam)
            env = pParam->pdflt;
    }
    if(!env || !*env)
        return false;
    if(val)
        *val = env;
    return true;
}
```

Reply:
looks good.